backend/app/policies/loan_policy.rb

```diff
@@ -1,45 +1,44 @@
 class LoanPolicy < ApplicationPolicy
   attr_reader :user, :loan
+
+  # Defines the scope of the policy, admins are granted access to everything and 
+  # Other People are only entitled to objects that they have created based on loaned_ny
   class Scope < Scope
     def resolve
-      if user.admin?
+      if user&.admin?
         scope.all
       else
         scope.where("loaned_by_id": user.id)
       end
     end
   end
 
+  # This is the policy for creating loans, you either have to be logged in or be a admin in 
+  # orde to create a loan
+
   def create?
-    unless user && user.admin? && user.uid == loan.lent_by_id
-      loan.errors.add(:lent_by, "authorization failed")
-      return false
-    end
-    return true
+    user || user&.admin?
   end
 
+  # This is the policy for updating loans, you either have to be an admin or the person who
+  # orignally created the loan in order to update it
+
   def update?
-    unless user&.admin? && user.uid == loan.lent_by_id
-      loan.errors.add(:lent_by, "authorization failed")
-      return false
-    end
-    return true
+    user&.admin? || user.uid == record.lent_by_id
   end
 
+  # This is the policy for showing all different loans, you either have to be an admin or the 
+  # person who created the loan in order to see the information about it
+
   def show?
-    unless user && user&.admin? && user.uid == loan.lent_by_id
-      loan.errors.add(:lent_by, "authorization failed")
-      return false
-    end
-    return true
+    user&.admin? || user.uid == record.lent_by_id
```

Comment:
```suggestion
    user&.admin? || user&.uid == record.lent_by_id
```
For safetys sake, maybe we should consider using the safe navigator operator on all of the user calls as a standard? What do you think? :)